tests/0143-exponential_backoff.c

```diff
@@ -0,0 +1,418 @@
+ * We fail the request with RD_KAFKA_RESP_ERR_NOT_LEADER_OR_FOLLOWER so that it triggers FastLeaderQuery request
+ * which is backed off exponentially
+*/
+void test_Produce_FastleaderQuery(rd_kafka_mock_cluster_t *mcluster,const char *topic, rd_kafka_conf_t *conf){
+        rd_kafka_mock_request_t **requests = NULL;
+        size_t request_cnt = 0;
+        int64_t previous_request_ts = -1;
+        int32_t retry_count = 0;
+        rd_bool_t produced = rd_false;
+        SUB_TEST();
+        test_conf_set(conf, "topic.metadata.refresh.interval.ms","-1");
+        rd_kafka_conf_set_dr_msg_cb(conf, test_dr_msg_cb);
+
+        rd_kafka_t *producer = test_create_handle(RD_KAFKA_PRODUCER, rd_kafka_conf_dup(conf));
+        rd_kafka_topic_t *rkt = test_create_producer_topic(producer, topic, NULL);
+        
+        rd_kafka_mock_push_request_errors(mcluster, RD_KAFKAP_Produce,  1,
+                                            RD_KAFKA_RESP_ERR_NOT_LEADER_OR_FOLLOWER);
+        rd_kafka_mock_clear_requests(mcluster);
+        test_produce_msgs(producer, rkt, 0, RD_KAFKA_PARTITION_UA, 0, 1, "hello",1);
+        rd_sleep(10);
+        requests = rd_kafka_mock_get_requests(mcluster,&request_cnt);
+        
+        for(size_t i = 0; i < request_cnt; i++){
+                TEST_SAY("Broker Id : %d API Key : %d Timestamp : %lld\n",rd_kafka_mock_request_id(requests[i]), rd_kafka_mock_request_api_key(requests[i]), rd_kafka_mock_request_timestamp(requests[i]));
+                if(!produced && rd_kafka_mock_request_api_key(requests[i]) == RD_KAFKAP_Produce)
+                        produced = rd_true;
+                if(rd_kafka_mock_request_api_key(requests[i]) == RD_KAFKAP_Metadata && produced){
+                        if(previous_request_ts != -1){
+                                int64_t time_difference = (rd_kafka_mock_request_timestamp(requests[i]) - previous_request_ts)/1000;
+                                int64_t low = ((1<<retry_count)*(retry_ms)*75)/100; 
+                                int64_t high = ((1<<retry_count)*(retry_ms)*125)/100;
+                                if (high > ((retry_max_ms*125)/100))
+                                        high = (retry_max_ms*125)/100;
+                                if (low > ((retry_max_ms*75)/100))
+                                        low = (retry_max_ms*75)/100;
+                                TEST_ASSERT((time_difference < high) && (time_difference > low),"Time difference is not respected!\n");
+                                retry_count++;
+                        }
+                        previous_request_ts = rd_kafka_mock_request_timestamp(requests[i]);
+                }
+        }
+        rd_kafka_topic_destroy(rkt);
+        rd_kafka_destroy(producer); 
+        rd_kafka_mock_clear_requests(mcluster);
+        SUB_TEST_PASS();
+}
+
+/** Fetch-FastLeaderQuery Test
+ * We fail the request with leader change or epoch to be precise so that it triggers FastLeaderQuery request.
```

Comment:
Comment needs more detail, like 

"We fail a Fetch request by causing a leader change (the leader is the same, but with a different leader epoch). It triggers fast leader query (Metadata request).
The request is able to obtain an updated leader, and hence, the fast leader query terminates after one Metadata request."